File: app/src/main/java/ca/gbc/comp3074/snapcal/data/model/Meal.kt
package ca.gbc.comp3074.snapcal.data.model

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey


@Entity(
    tableName = "meals",
    indices = [
        Index(value = ["createdAt"]),
        Index(value = ["mealType"])
    ]
)
data class Meal(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0L,


    val name: String,
    val calories: Int,
    val protein: Float,
    val carbs: Float,
    val fat: Float,


    val mealType: String,       // "breakfast" | "lunch" | "dinner" | "snack"
    val createdAt: Long,

    val portionGrams: Float? = null,
    val photoPath: String? = null,
    val notes: String? = null
)

File: SnapCal_iOS/SnapCal_iOS/SnapCal_iOS/Models/Meal.swift
import Foundation
import SwiftData

@Model
final class Meal {
    var id: UUID
    var name: String
    var calories: Int
    var protein: Double
    var carbs: Double
    var fat: Double
    var mealType: String
    var createdAt: Date
    var portionGrams: Double?
    var photoPath: String?
    var notes: String?

    init(
        id: UUID = UUID(),
        name: String,
        calories: Int,
        protein: Double,
        carbs: Double,
        fat: Double,
        mealType: String,
        createdAt: Date = .now,
        portionGrams: Double? = nil,
        photoPath: String? = nil,
        notes: String? = nil
    ) {
        self.id = id
        self.name = name
        self.calories = calories
        self.protein = protein
        self.carbs = carbs
        self.fat = fat
        self.mealType = mealType
        self.createdAt = createdAt
        self.portionGrams = portionGrams
        self.photoPath = photoPath
        self.notes = notes
    }
}
